lib/src/view/broadcast/broadcast_analysis_screen.dart

```diff
@@ -0,0 +1,408 @@
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/duration.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/utils/lichess_assets.dart';
+import 'package:lichess_mobile/src/view/analysis/analysis_board.dart';
+import 'package:lichess_mobile/src/view/analysis/analysis_screen_body.dart';
+import 'package:lichess_mobile/src/view/analysis/analysis_settings.dart';
+import 'package:lichess_mobile/src/view/analysis/engine_depth.dart';
+import 'package:lichess_mobile/src/widgets/adaptive_bottom_sheet.dart';
+import 'package:lichess_mobile/src/widgets/buttons.dart';
+import 'package:lichess_mobile/src/widgets/platform.dart';
+
+class BroadcastAnalysisScreen extends ConsumerWidget {
+  final BroadcastRoundId roundId;
+  final BroadcastGameId gameId;
+  final String title;
+
+  const BroadcastAnalysisScreen({
+    required this.roundId,
+    required this.gameId,
+    required this.title,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return ConsumerPlatformWidget(
+      androidBuilder: _androidBuilder,
+      iosBuilder: _iosBuilder,
+      ref: ref,
+    );
+  }
+
+  Widget broadcastWrapBoardBuilder(
+    WidgetRef ref,
+    AnalysisBoard board,
+    AnalysisControllerProvider ctrlProvider,
+    double boardSize,
+  ) {
+    final clocks = ref.watch(ctrlProvider.select((value) => value.clocks));
+    final currentPath = ref.watch(
+      ctrlProvider.select((value) => value.currentPath),
+    );
+    final broadcastLivePath = ref.watch(
+      ctrlProvider.select((value) => value.broadcastLivePath),
+    );
+    final playingSide =
+        ref.watch(ctrlProvider.select((value) => value.position.turn));
+    final game = ref.watch(
+      broadcastRoundControllerProvider(roundId)
+          .select((game) => game.value?[gameId]),
+    );
+    final pov = ref.watch(ctrlProvider.select((value) => value.pov));
+
+    return Column(
+      children: [
+        if (game != null)
+          _PlayerWidget(
+            clock: (playingSide == pov.opposite)
+                ? clocks?.parentClock
+                : clocks?.clock,
+            width: boardSize,
+            game: game,
+            side: pov.opposite,
+            boardSide: _PlayerWidgetSide.top,
+            playingSide: playingSide,
+            playClock: currentPath == broadcastLivePath,
+          ),
+        board,
+        if (game != null)
+          _PlayerWidget(
+            clock: (playingSide == pov) ? clocks?.parentClock : clocks?.clock,
+            width: boardSize,
+            game: game,
+            side: pov,
+            boardSide: _PlayerWidgetSide.bottom,
+            playingSide: playingSide,
+            playClock: currentPath == broadcastLivePath,
+          ),
+      ],
+    );
+  }
+
+  Widget _androidBuilder(BuildContext context, WidgetRef ref) {
+    final pgn = ref.watch(
```

Comment:
Here again this is rather confusing. We're seeing a ref.watch, but after investigation I realise the pgn is never changing. It's only here to feed the analysis controller.

This is indicating something is wrong in the architecture. I told you in our last conversation that it was ok to keep the analysis controller and we'd refactor later, but I changed my mind. If there wasn't the existing analysis controller, I'm pretty sure you'd have done it differently, because without the prior implementation it would have made much more sense to do like what I suggest below:

A `BroadcastGameController` with this signature:
```dart
@riverpod
class BroadcastGameController extends $BroadcastGameController {
   Future<BroadcastGameState> build(BroadcastGameId id) {
      ....
      // here you get and watch the game through http / websocket
   }
}
```

In the broadcast screen you only watch this controller to get changes (as opposed to 3 different ones right now).
The live clock in the `broadcast_analysis_screen` must be implemented with a `CountdownClock` widget (appearance may need adjustment). It should work like it works right now in the TV screen.

It is ok to copy/paste all the logic from analysis controller for now (I'll work on mixins later for shared logic). It should look like the new `StudyController` which has a similar signature.

Reply:
The pgn is changing. On each move received from the websocket I edit the pgn to add the new move.